listener.py

```
@@ -296,19 +303,35 @@ def process_pending(cfg: dict):
-                mode=mode,
-                railway_url=railway_url,
-                api_key=api_key,
-                interactive=False,   # never prompt for input when running via listener
-            )
+            # Run scrape in a thread so we can enforce a hard 5-minute wall-clock
+            # timeout.  A hung Playwright call would otherwise block the whole queue.
+            JOB_TIMEOUT = 300  # seconds
+
+            def _do_scrape():
+                return scrape(
+                    group,
```

Ruff (0.15.2)

[warning] Function definition does not bind loop variable `group`

(B023)